grid_map_core/src/GridMap.cpp

```diff
@@ -153,116 +173,137 @@ bool GridMap::erase(const std::string& layer) {
   return true;
 }
 
-const std::vector<std::string>& GridMap::getLayers() const {
+const std::vector<std::string> & GridMap::getLayers() const
+{
   return layers_;
 }
 
-float& GridMap::atPosition(const std::string& layer, const Position& position) {
+float & GridMap::atPosition(const std::string & layer, const Position & position)
+{
   Index index;
   if (getIndex(position, index)) {
     return at(layer, index);
   }
   throw std::out_of_range("GridMap::atPosition(...) : Position is out of range.");
 }
 
-float GridMap::atPosition(const std::string& layer, const Position& position, InterpolationMethods interpolationMethod) const {
-
+float GridMap::atPosition(
+  const std::string & layer, const Position & position,
+  InterpolationMethods interpolationMethod) const
+{
   bool skipNextSwitchCase = false;
   switch (interpolationMethod) {
     case InterpolationMethods::INTER_CUBIC_CONVOLUTION: {
-      float value;
-      if (atPositionBicubicConvolutionInterpolated(layer, position, value)) {
-        return value;
-      } else {
-        interpolationMethod = InterpolationMethods::INTER_LINEAR;
-        skipNextSwitchCase = true;
-      }
-    }
-    case InterpolationMethods::INTER_CUBIC: {
-      if (!skipNextSwitchCase) {
         float value;
-        if (atPositionBicubicInterpolated(layer, position, value)) {
+        if (atPositionBicubicConvolutionInterpolated(layer, position, value)) {
           return value;
         } else {
           interpolationMethod = InterpolationMethods::INTER_LINEAR;
+          skipNextSwitchCase = true;
         }
+        [[fallthrough]];
+      }
+    case InterpolationMethods::INTER_CUBIC: {
+        if (!skipNextSwitchCase) {
+          float value;
+          if (atPositionBicubicInterpolated(layer, position, value)) {
+            return value;
+          } else {
+            interpolationMethod = InterpolationMethods::INTER_LINEAR;
+          }
+        }
+        [[fallthrough]];
       }
-    }
     case InterpolationMethods::INTER_LINEAR: {
-      float value;
-      if (atPositionLinearInterpolated(layer, position, value))
-        return value;
-      else
-        interpolationMethod = InterpolationMethods::INTER_NEAREST;
-    }
+        float value;
+        if (atPositionLinearInterpolated(layer, position, value)) {
+          return value;
+        } else {
+          interpolationMethod = InterpolationMethods::INTER_NEAREST;
+        }
+        [[fallthrough]];
```

Comment:
using [fallthrough](https://en.cppreference.com/w/cpp/language/attributes/fallthrough) (C++17, is that okay?) I think the logic relies on not having break statements at the end of the cases, but the compiler was not happy with with that.

Reply:
You should use C++14 unless absolutely required to use C++17 - ROS2 Foxy is under C++14.